### PR TITLE
Remove usage of QuasiQuotes

### LINKED ISSUE #192: directory master uses quasiquotes
https://github.com/haskell/directory/commit/367241f977b102a394be348f176c9688f1130f42 introduced a usage of QuasiQuotes which is not allowed for core libraries (some platforms do not support TH).

### PR DESCRIPTION
**Pull request by wz1000**
Remove usage of QuasiQuotes

This is necessary to allow directory to be build with GHC

Fixes #192

### CODE REVIEW

File: System/Directory/OsPath.hs [hunk 1]
```diff
@@ -751,7 +752,13 @@ withReplacementFile :: OsPath            -- ^ Destination file
 withReplacementFile path postAction action =
   (`ioeAddLocation` "withReplacementFile") `modifyIOError` do
     mask $ \ restore -> do
-      (tmpFPath, hTmp) <- OS.openBinaryTempFile (takeDirectory path) [osstr|.copyFile.tmp|]
+      let tmpPath = case encodeWith (mkUTF8 ErrorOnCodingFailure)
+                                    (mkUTF16le ErrorOnCodingFailure)
+                                    ".copyFile.tmp"
+                         of
+            Left err -> error ("withReplacementFile: invalid encoding: " ++ show err)
```
Comment (hasufell) [inline System/Directory/OsPath.hs:759]: `error` is a bit depressing, why don't we throw an IO error?

Comment (wz1000) [reply System/Directory/OsPath.hs:759]: It's not something that should really happen under any usage of the library, so I don't see the benefit of `throwIO` which makes it easier to catch.

**Comment by wz1000** [comment]
@hasufell any updates here?

**Comment by hasufell** [comment]
I'm not a maintainer, just commenting, because I was involved in the code.

**Comment by wz1000** [comment]
ping @Rufflewind

**Pull request merged**